internal/cmdsearch/cmdsearch.go

```diff
@@ -109,6 +112,7 @@ func (r *SearchRunner) ExecuteCmd(w io.Writer, pkgPath string) error {
 		PutLiteral:   r.PutLiteral,
 		PutPattern:   r.PutPattern,
 		PackagePath:  pkgPath,
+		Validate:     r.Validate,
```

Comment:
Does the validation step need to be part of `SearchReplace`? It seems like it is basically a step we perform after we have made the change. Could we implement this as a separate step after we have run SearchReplace? It would make the solution more modular and it would prevent adding too much functionality into `SearchReplace`. 

I'm also wondering if this is something that could be supported with a function instead of adding the functionality directly into kpt.

Reply:
`It seems like it is basically a step we perform after we have made the change` => We throw an error and not write the result back if there are errors. So it is a part of Search and Replace operation. 

We can support this as a function but this feature would need the validation. By including this option in Search and Replace, users can get immediate feedback when they replace field values.